File: spark_sql,ml/wordCount.java
package com.dt.sparkapps.cores; 
import org.apache.SparkConf;
import org.apache.spark.api.java.JavaSparkContext;
public class WordCount{
	public void main(String[] arsgs){
		
		// 应用java 开发spark 程序着实操蛋 ，还是学好scala，开发spark ，flink 等程序 
		SparkConf = conf  = new SparkConf().setAppName("spark WordCount by java !").setMaster("local");
		javaSparkContext sc = new javaSparkContext(conf); // 底层其实是scala sc 的封装 ；
		JavaRDD<String> textFile = sc.textFile("hdfs://...");
		JavaRDD<String> words = textFile.flatMap(new FlatMapFunction<String, String>() {
   		public Iterable<String> call(String s) { 
   			return Arrays.asList(s.split(" ")); }
 		});
		JavaPairRDD<String, Integer> pairs = words.mapToPair(new PairFunction<String, String, Integer>() {
   			public Tuple2<String, Integer> call(String s) 
   			{ return new Tuple2<String, Integer>(s, 1); }
 		});
		JavaPairRDD<String, Integer> counts = pairs.reduceByKey(new Function2<Integer, Integer, Integer>() {
   			public Integer call(Integer a, Integer b) { return a + b; }
 		});
 		counts.saveAsTextFile("hdfs://..."); 

	}
}

File: spark_sql,ml/WordCount.scala
package com.dt.spark

import org.apache.spark.SparkConf 
object WordCount{
	def main(args:Array[String]){
		/**
		* 创建SparkConf 的配置对象
		*/
		val conf = new SparkConf();
		conf.setAppName("MY first Spark ")
		conf.setMaster("local")// 此时，程序在本地执行，不需要spark 集群
		// conf.setMaster("spark:// Master:7077 ")
		// 创建SparkContext 对象,是spark程序的唯一入口，
		val sc = new SparkContext(conf)
		// 创建RDD ，数据会被RDD 分成一系列的partitions，分配到每个partitions的数据属于一个task 的处理范畴，，
		val lines:RDD[String] = sc.textFile("D//Big_Data")
		// 对RDD 进行transformation 处理 map，filter 等高阶函数编程 ，

		val words = lines.flatMap(_.split(" "))
		val pairs = words.map(_=>(_,1))
		val wordCount = pairs.reduceByKey(_+_)
		wordCount.foreach(pairs=>println(pairs._1+":"+pairs._2))
	}
}